src/wp-includes/rest-api/endpoints/class-wp-rest-post-types-controller.php

```diff
@@ -221,6 +221,10 @@ public function prepare_item_for_response( $item, $request ) {
 			$data['slug'] = $post_type->name;
 		}
 
+		if ( in_array( 'icon', $fields, true ) ) {
+			$data['icon'] = $post_type->menu_icon;
```

Comment:
If this value is null, should it maybe default to something?

Reply:
I think it's okay to be null since it's not required and the consumers can decide what to do with it. For example in php core for post types we assign some default icon if it's in the `admin menu`, but in the site editor we assign different default icon..